# Baseline

In [4]:
import pandas as pd
from IPython.display import display


# Load the data
file_path = r"../data/raw/ACME-HappinessSurvey2020.csv"
df = pd.read_csv(file_path)

# Display basic information about the dataset
display(df.info())

# Display the first few rows of the dataset
df.sample(5)


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 126 entries, 0 to 125
Data columns (total 7 columns):
 #   Column  Non-Null Count  Dtype
---  ------  --------------  -----
 0   Y       126 non-null    int64
 1   X1      126 non-null    int64
 2   X2      126 non-null    int64
 3   X3      126 non-null    int64
 4   X4      126 non-null    int64
 5   X5      126 non-null    int64
 6   X6      126 non-null    int64
dtypes: int64(7)
memory usage: 7.0 KB


None

Y  X1  X2  X3  X4  X5  X6
119  1   5   5   1   1   5   1
78   0   5   3   3   3   2   5
69   1   5   4   5   5   5   5
34   1   4   1   3   1   1   4
86   1   5   3   3   3   5   5

In [5]:
fcols = df.select_dtypes('float').columns
icols = df.select_dtypes('integer').columns

df[fcols] = df[fcols].apply(pd.to_numeric, downcast='float')
df[icols] = df[icols].apply(pd.to_numeric, downcast='integer')

In [8]:
df.to_parquet("../data/clean/ACME-happinesSurvey2020.parquet", compression="brotli")

df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 126 entries, 0 to 125
Data columns (total 7 columns):
 #   Column  Non-Null Count  Dtype
---  ------  --------------  -----
 0   Y       126 non-null    int8 
 1   X1      126 non-null    int8 
 2   X2      126 non-null    int8 
 3   X3      126 non-null    int8 
 4   X4      126 non-null    int8 
 5   X5      126 non-null    int8 
 6   X6      126 non-null    int8 
dtypes: int8(7)
memory usage: 1010.0 bytes


The data has been successfully loaded and consists of 126 entries with 7 columns: `Y`, `X1`, `X2`, `X3`, `X4`, `X5`, and `X6`. There are no missing values in the dataset.

### Next Steps:

1. **Data Preprocessing**:
   - Since the data is already clean, we can proceed with splitting it into training and test sets.
2. **Feature Selection**:
   - We'll use techniques like Recursive Feature Elimination (RFE) to identify the most important features.
3. **Model Training and Evaluation**:
   - Train a classification model (e.g., Logistic Regression, Random Forest).
   - Evaluate the model using accuracy and other metrics.